src/main/java/net/openhft/chronicle/wire/AbstractWire.java

```diff
@@ -310,6 +310,8 @@ public long enterHeader(long safeLength) {
             if (isNotComplete(header)) {
                 if (header != END_OF_DATA)
                     Jvm.warn().on(getClass(), new Exception("Incomplete header found at pos: " + pos + ": " + Integer.toHexString(header) + ", overwriting"));
+                else
+                    throw new IllegalStateException("You should not be able to write at EOF");
```

Comment:
enterHeader's behaviour is tested in CQ

Reply:
Should we have a specific subclass of IllegalStateException? it seems like a pretty specific scenario

Reply:
Done